packages/core/nano/src/types.ts

```diff
@@ -256,15 +274,27 @@ export interface IMatrixReader<T> {
     readonly rowCount: number;
     readonly colCount: number;
     getCell(row: number, col: number): T;
+
+    /**
+     * A reference to the underlying matrix producer that provides values for this reader.
+     */
+    readonly matrixProducer: IMatrixProducer<T>;
```

Comment:
Is it worth making this a function that can capture a reference, hopefully avoiding the need to defined as `undefined` and then reassign.

Another thing to consider in the future is how this gets used. If it's primarily for closing the reader, then could we have the reader support `close`, maybe even capturing the consumer so it doesn't need to be supplied? (Though then we need a reader per consumer). I don't think needs to be changed in this PR.

Reply:
> Is it worth making this a function that can capture a reference

Oh yeah...  You can implement a readonly field with a property accessor (e.g., `get matrixProducer() { ... }`).  So my minor pain point is addressed.

> Another thing to consider in the future is how this gets used.

My other use is for comparing the incoming `producer` during change notifications to figure out which of my dependencies has changed.